content/articles/implementing-multiple-viewholder-classes-in-android-using-kotlin/index.md

```diff
@@ -0,0 +1,349 @@
+
+```kotlin
+override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    if (getItemViewType(position) == HASIMAGE){
+        (holder as LandmarkWithImageViewHolder).bind(landmarks[position])
+    } else{
+        (holder as LandmarkWithoutImageViewHolder).bind(landmarks[position])
+    }
+}
+```
+
+### Generating data
+The following is dummy data used for demonstration purposes. Ideally, you should retrieve structured data from a real database.
+
+```kotlin
+class LandmarkModel {
+    companion object {
+        fun getLandmarks(): ArrayList<Landmark> = arrayListOf(
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 1", null, HasImage.FALSE),
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 2", null, HasImage.FALSE),
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 3", null, HasImage.FALSE),
+            Landmark(
+                "Mt. Kenya", "This is a mountain in Kenya 4", R.drawable.ic_launcher_background,
+                HasImage.TRUE
+            ),
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 5", null, HasImage.FALSE),
+            Landmark(
+                "Mt. Kenya", "This is a mountain in Kenya 6", R.drawable.ic_launcher_foreground,
+                HasImage.TRUE
+            ),
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 7", null, HasImage.FALSE),
+            Landmark("Mt. Kenya", "This is a mountain in Kenya 8", null, HasImage.FALSE)
+        )
+    }
+}
+```
+
+The images referenced above, are generated by default when starting a project. You can use your images of choice as well.
+
+### populating the RecyclerView
+The final step is to show this data to the user.
+
+In the **MainActivity.kt** file, paste the following code:
+
+```kotlin
+class MainActivity : AppCompatActivity() {
+    private lateinit var mainBinding: ActivityMainBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // remember to set this value to null in the onDestroy method to avoid memory leaks.
+        mainBinding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(mainBinding.root)
+        
+        val landmarks = LandmarkModel.getLandmarks()
+        mainBinding.landmarkRecyclerview.apply {
+            adapter = LandmarkAdapter(landmarks)
+        }
+    }
+}
+```
+
+### Testing the App
+upon running the app, you should expect to see something similar to this.
+
+![Test App](test-app.png)
+
+### Conclusion
+In this tutorial, we have learned the fundamental concepts of displaying different types of items in a single recyclerView with the use of multiple viewHolders. The knowledge gained in this tutorial can be applied to other more sophisticated use cases with the same goal.
+
+You can get the full code implementation in this [github repository](https://github.com/MaurineM/multiple-view-holders).
```

Comment:
This repo is currently empty or private. Kindly update it and ensure it is `public`.